### DISCUSSION TITLE
Converting BStr/BString to std::String

### DISCUSSION THREAD

**Discussion by RAnders00** [BurntSushi/bstr#172] [Q&A] [answered]
I think this is a common use case. I can't find anything in the documentation. How is this done?

**Comment by RAnders00** [comment]
`try_into()`/`String::try_from()` should do the trick: https://github.com/BurntSushi/bstr/blob/ec2154a36943455c22de64331cfc1624ba93ec3c/src/impls.rs#L264-L271

(Why is documentation not mentioning this? :( )

**Reply by BurntSushi** [reply]
The documentation does mention it: https://docs.rs/bstr/latest/bstr/struct.BString.html#impl-TryFrom%3CBString%3E-for-String

You need to know to look for `TryFrom` impls.

The prose docs do not include a refresher on `TryFrom` since it's a foundational trait in the standard library.

On top of that, there are `ByteSlice::to_str` and `ByteVec::into_string` methods that one can discover by looking at the methods on the core traits in this crate.

**Comment by BurntSushi** [comment, marked as answer]
A `BStr` can be converted to a `&str` via [`ByteSlice::to_str`](https://docs.rs/bstr/latest/bstr/trait.ByteSlice.html#method.to_str). It's one of the first methods listed on the trait. And if `s` is a `&str`, you can get a `String` via `s.to_string()`.

A `BString` can be converted to a `String` via [`ByteVec::into_string`](https://docs.rs/bstr/latest/bstr/trait.ByteVec.html#method.into_string).

There are also `TryFrom` impls for [`TryFrom<&BString> for &str`](https://docs.rs/bstr/latest/bstr/struct.BString.html#impl-TryFrom%3C%26'a+BString%3E-for-%26'a+str) and [`TryFrom<BString> for String`](https://docs.rs/bstr/latest/bstr/struct.BString.html#impl-TryFrom%3CBString%3E-for-String). Analogously, there is also [`TryFrom<&BStr> for &str`](https://docs.rs/bstr/latest/bstr/struct.BStr.html#impl-TryFrom%3C%26'a+BStr%3E-for-%26'a+str) and [`TryFrom<&BStr> for String`](https://docs.rs/bstr/latest/bstr/struct.BStr.html#impl-TryFrom%3C%26'a+BStr%3E-for-String).
